### DISCUSSION TITLE
Jellyfin nginx Syncplay

### DISCUSSION THREAD

**Discussion by GarrusVaka** [jellyfin/jellyfin#9395] [General]
Hello,

I would like to make my Jellyfin server externally accessible (also works) but I would like to use SSL (also works).

However, as soon as I want to use the Syncplay feature, it does not work. But it works without problems via http. Can anyone find someone who is willing to help me?

I use nginx and Let's Encrypt on Linux Debian. 

Attached is my domain config for nginx.

[stream.DOMAIN.tld.txt](https://github.com/jellyfin/jellyfin/files/10830977/stream.DOMAIN.tld.txt)

**Comment by Zovereign** [comment]
What I did and worked for me is, I use nginx-proxy-manager (NPM).
Default port 80 and 443 mapped to this NPM.
And nginx alone that I have, I change to some port else.

Then let the NPM do the rest of the job for me including those ssl tasks.

**Reply by Ferk** [reply]
@Zovereign thanks! ...do you have any insight about what NPM is doing different?

If there's an option in NPM to see what's the nginx configuration it's setting, it might be helpful to see. It feels kind of wasteful to set up a new docker with NPM just for this.

**Comment by Ferk** [comment]
I have the same problem.. in addition, I have set it up to use a subpath (`example.com/media`) instead of a subdomain (`media.example.com/`).

Below is the relevant section of my config, which is set up following [what was instructed here](https://jellyfin.org/docs/general/networking/nginx/):
```sh
server {
    listen       443 ssl;
   # ... -snip-

    location /media/ {
        proxy_pass http://$myserver:8096;
        proxy_pass_request_headers on;

        proxy_set_header Host $host;
        proxy_set_header X-Real-IP $remote_addr;
        proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
        proxy_set_header X-Forwarded-Proto $scheme;
        proxy_set_header X-Forwarded-Host $http_host;

        # Websocket support
        proxy_set_header Upgrade $http_upgrade;
        proxy_set_header Connection $http_connection;

        # Disable buffering when the nginx proxy gets very resource heavy upon streaming
        proxy_buffering off;
    }
}
```

**Comment by tydog98** [comment]
For Syncplay to work you need Websockets support enabled

**Reply by TimGels** [reply]
This is a really important thing for SyncPlay yes. Make sure websockets are also working when setting up a reverse proxy for Jellyfin.
